users/uqs/rules.mk

```
@@ -0,0 +1,21 @@
+# don't include for keyboards/ploopyco/mouse/keymaps/uqs
+ifeq ($(filter $(strip $(KEYBOARD)),ploopyco/mouse),)
+SRC += uqs.c
+
+RGBLIGHT_ENABLE ?= yes      # Enable keyboard RGB underglow
+LTO_ENABLE ?= yes  # disables the legacy TMK Macros and Functions features
+
+UCIS_ENABLE = yes
+LEADER_ENABLE = yes
+COMBO_ENABLE = yes
+MOUSEKEY_ENABLE = yes
+
+# Disable all the unused stuff.
+SPACE_CADET_ENABLE = no
+COMMAND_ENABLE = no
+MAGIC_ENABLE = no
+
+# From https://michael.stapelberg.ch/posts/2021-05-08-keyboard-input-latency-qmk-kinesis/
+# adds about 900 bytes!
+DEBOUNCE_TYPE = asym_eager_defer_pk
```

```suggestion
    SRC += uqs.c

    RGBLIGHT_ENABLE ?= yes      # Enable keyboard RGB underglow
    LTO_ENABLE ?= yes  # disables the legacy TMK Macros and Functions features

    UCIS_ENABLE = yes
    LEADER_ENABLE = yes
    COMBO_ENABLE = yes
    MOUSEKEY_ENABLE = yes

    # Disable all the unused stuff.
    SPACE_CADET_ENABLE = no
    COMMAND_ENABLE = no
    MAGIC_ENABLE = no

    # From https://michael.stapelberg.ch/posts/2021-05-08-keyboard-input-latency-qmk-kinesis/
    # adds about 900 bytes!
    DEBOUNCE_TYPE = asym_eager_defer_pk
```

Also, I would NOT recommend the asym eager defer pk algo for non-arm controllers, IIRC.

---

care to explain? is it too slow on the AVRs? which ones are suitable and faster for AVR?

---

I'd have to dig up the conversation, but yeah, it's slower and more computationally expensive on AVR.

---

hmpf, how to make this conditional in my users/uqs/rules.mk though? It seems to get read before $(MCU) was set :/

---

post_rules.mk may work, but I'm not sure.